action.yml

```diff
@@ -48,11 +48,29 @@ runs:
         $platform = $platform -replace '^i686$', 'x86'
         # validate that platform is one of the expected values
         if (($platform -ne 'x86') -and ($platform -ne 'x86_64')) {
-          echo "unknown platform $platform"
-          exit 1
+          throw "Unknown platform $platform."
         }
         $setupFileName = "setup-$platform.exe"
-        Invoke-WebRequest "https://cygwin.com/$setupFileName" -OutFile C:\setup.exe
+
+        $maxRetries = 3
+        $retryCount = 0
+        $success = $false
+
+        while (-not $success -and $retryCount -lt $maxRetries) {
+          try {
+            Invoke-WebRequest -Uri "https://cygwin.com/$setupFileName" -OutFile "C:\setup.exe"
+            $success = $true
+          } catch {
+            Write-Output "Attempt $($retryCount + 1) failed. Retrying..."
+            Start-Sleep -Seconds 10
```

Comment:
What about starting lower, like 5 seconds, which is the PowerShell default interval (https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/invoke-webrequest?view=powershell-7.4#-retryintervalsec)?

And perhaps cmobine it with exponential backoff on subsequent retries? Though with 3 retries, this may be overkill.